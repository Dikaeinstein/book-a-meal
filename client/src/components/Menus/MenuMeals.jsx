import React from 'react';
import PropTypes from 'prop-types';

const MenuMeals = ({ meals }) => {
  const renderMealList = () => meals.map(meal => (
    <tr key={String(meal.id)}>
      <td>{meal.name}</td>
      <td>&#x20a6; {meal.price}</td>
    </tr>
  ));

  return (
    <table className="font-weight-bold">
      <tbody>
        {renderMealList()}
      </tbody>
    </table>
  );
};

MenuMeals.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default MenuMeals;
