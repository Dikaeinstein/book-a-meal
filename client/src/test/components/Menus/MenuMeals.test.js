import React from 'react';
import { shallow } from 'enzyme';
import MenuMeals from '../../../components/Menus/MenuMeals';


describe('<MenuMeals />', () => {
  const meals = [
    { id: 1, name: 'test meal', price: 4000 },
    { id: 2, name: 'test meal', price: 4000 },
  ];

  it('should match snapshot', () => {
    const tree = shallow(<MenuMeals meals={meals} />);
    expect(tree).toMatchSnapshot();
  });
});
