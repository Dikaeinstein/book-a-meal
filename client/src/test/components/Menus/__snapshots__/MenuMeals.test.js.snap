// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuMeals /> should match snapshot 1`] = `
<table
  className="font-weight-bold"
>
  <tbody>
    <tr
      key="1"
    >
      <td>
        test meal
      </td>
      <td>
        ₦ 
        4000
      </td>
    </tr>
    <tr
      key="2"
    >
      <td>
        test meal
      </td>
      <td>
        ₦ 
        4000
      </td>
    </tr>
  </tbody>
</table>
`;
